File: FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/moeglobal/slam/SlamHandler.java
package org.firstinspires.ftc.robotcontroller.moeglobal.slam;

import android.content.Context;
import android.content.IntentFilter;
import android.hardware.usb.UsbDevice;
import android.hardware.usb.UsbManager;
import android.util.Log;

import static android.hardware.usb.UsbManager.ACTION_USB_DEVICE_ATTACHED;
import static android.hardware.usb.UsbManager.ACTION_USB_DEVICE_DETACHED;
import static org.firstinspires.ftc.robotcontroller.moeglobal.slam.SlamUsbListener.handleLizardDeviceAdded;

public class SlamHandler {
    public static volatile SlamT265Handler t265Handler;
    public static IntentFilter USBAddOrRemove = new IntentFilter();
//    private static SlamUsbListener usbReceiver = new SlamUsbListener();

    static {
        USBAddOrRemove.addAction(ACTION_USB_DEVICE_ATTACHED);
        USBAddOrRemove.addAction(ACTION_USB_DEVICE_DETACHED);
    }

    public static void init(Context context) {
        SlamUsbHandler.init(context);
        SlamLizardHandler.init(context);
        checkPreConnection();
    }

    /**
     * checks if devices are already connected
     */
    private static void checkPreConnection() {
        Log.e("checking", "connection");
        UsbDevice device = SlamLizardHandler.getDevice();
        if (device != null) {
            SlamUsbListener.handleLizardDeviceAdded(device);
            Log.e("slam", "lizard");
        } else {
            Log.e("no lizard", "np");

            device = SlamT265Handler.Companion.getDevice();
            if (device != null) {
                Log.e("slam", "legit");

                SlamUsbListener.handleT265DeviceAdded(device);
            }
        }
    }

//    public static void initUsbListener(Context context) {
//        context.registerReceiver(usbReceiver, USBAddOrRemove);
//    }

//    public static void unRegisterListener(Context context) {
//        context.unregisterReceiver(usbReceiver);
//    }

    public static void handleDeviceAdded(UsbDevice usbDevice) {
        SlamUsbListener.handleDeviceAdded(usbDevice);

    }

    public static void handleDeviceRemoved(UsbDevice usbDevice) {
        SlamUsbListener.handleDeviceRemoved(usbDevice);
    }
}

File: FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/moeglobal/slam/SlamHandler.kt
package org.firstinspires.ftc.robotcontroller.moeglobal.slam

import android.content.Context
import android.content.IntentFilter
import android.hardware.usb.UsbManager
import android.util.Log
import org.firstinspires.ftc.robotcontroller.internal.FtcRobotControllerActivity

object SlamHandler {
    @Volatile
    var t265Handler: SlamT265Handler? = null
    private val USBListener = SlamUSBListener()
    var USBAddOrRemove = IntentFilter().apply {
        addAction(UsbManager.ACTION_USB_DEVICE_ATTACHED)
        addAction(UsbManager.ACTION_USB_DEVICE_DETACHED)
    }


    @JvmStatic
    fun init(context: FtcRobotControllerActivity) {
        SlamUsbHandler.init(context)
        SlamLizardHandler.init(context)
        SlamT265Handler.init(context)
        checkPreConnection()
        initUsbListener(context)
    }

    /**
     * checks if devices are already connected
     */
    private fun checkPreConnection() {
        Log.e("checking", "connection")
        var device = SlamLizardHandler.getDevice()
        if (device != null) {
            SlamUsbReciever.handleLizardDeviceAdded(device)
            Log.e("slam", "lizard")
        } else {
            Log.e("no lizard", "np")
            device = SlamT265Handler.device
            if (device != null) {
                Log.e("slam", "legit")
                SlamUsbReciever.handleT265DeviceAdded(device)
            }
        }
    }

    private fun initUsbListener(context: Context) {
        context.registerReceiver(USBListener, USBAddOrRemove)
    }

    //    public static void unRegisterListener(Context context) {
    //        context.unregisterReceiver(usbReceiver);
    //    }
    //    @JvmStatic
    //    fun handleDeviceAdded(usbDevice: UsbDevice) {
    //        SlamUsbReciever.handleDeviceAdded(usbDevice)
    //    }
    //
    //    @JvmStatic
    //    fun handleDeviceRemoved(usbDevice: UsbDevice) {
    //        SlamUsbReciever.handleDeviceRemoved(usbDevice)
    //    }


}